### ISSUE TITLE
openshift environment and mpcm

### ISSUE THREAD

**Issue by NicoledeGreef** [bcgov/smk-legacy-ui#80]
There are challenges in having openshift hosted apps reach MPCM which is critical for SMK.

**Comment by dhlevi** [comment]
Some initial discussion, from the slack channel

Benjamin Jubb [9:12 AM]
The service is trying to access http://apps.gov.bc.ca/pub/mpcm/services/catalog/PROD, which is accessible from everywhere I’ve tried, except in the smk pod. If I hit that endpoint with curl in the terminal, I get a ‘Connection reset by peer’ error. ideas?

ll911 [9:12 AM]
that’s by design
http://apps.gov.bc.ca/* accessible from public only, not from inside

Benjamin Jubb [9:14 AM]
ok. whats the internal name?

ll911 [9:14 AM]
at the moment, not exposed
why need to exposed internal?

Benjamin Jubb [9:15 AM]
SMK uses that service
what are our options here? SMK needs access to that service somehow.

ll911 [10:16 AM]
can it be done on client side?
if access via browser, you will be ok
which is opposite for route planner, it needs to be server side calls.

Benjamin Jubb [10:28 AM]
in theory the client could access it, but that would require a major re-work of the app. realistically, right now, the SMK service needs to be able to get to that endpoint

**Comment by ll911** [comment]
this has nothing to do with how openshift setup, @NicoledeGreef you have false assumption, this is only related to how [apps.gov.bc.ca] setup

**Comment by cpspicer** [comment]
Ben - given the architecture limitations of apps.gov.bc.ca, can you estimate what it might take in terms of time to update the client to make the api call to MPCM?  Not sure if it would help, but we could offer up the old code in DMF if it at all might be useful.

**Comment by ll911** [comment]
@cpspicer you have two options, 
1, recode smk client piece consume mpcm to use client side traffic
2, expose mpcm service catalog via server end gateway, so the mpcm catalog will be exposed globally

**Comment by cpspicer** [comment]
@ll911 As above, I've asked Ben to cost option 1, as it will be the most expedient.

**Comment by qqnluaq** [comment]
For option #1, that kind of change to the admin UI implies a larger effort to remove the dependence on JSF, and make the admin UI operate only client-side, interacting with the SMK service. 
Estimate 5 days of work for this.

**Comment by cpspicer** [comment]
@qqnluaq Thanks - could you follow up with the MPCM upgrade team and see what kind of effort would be involved in an MPCM/gateway integration?  Need to review viability based on its current use pattern and compare.  Assume you'll need a decision on approach soon given current sprint planning.

Ty

**Comment by cboreen** [comment]
In chatting with Leo, we feel that we can easily expose the MPCM web service via the Kong gateway so we do not have to undertake this change right now and focus on more valuable tasks.  We would configure:

expose it as mpcm-catalogue.api.gov.bc.ca to apps.gov.bc.ca/pub/mpcm/services/

@cpspicer we will need your agreement with the approach.

**Comment by cpspicer** [comment]
I agree - glad we came up with an optimal option that conserves resources.  Thanks guys.

**Comment by ll911** [comment]
https://mpcm-catalogue.api.gov.bc.ca/ ready

**Comment by cboreen** [comment]
@qqnluaq please update the url and test it

**Comment by qqnluaq** [comment]
I tried 
`curl  'https://mpcm-catalogue.api.gov.bc.ca/pub/mpcm/services/catalog/PROD/'`
and got
`<html><body>No service was found.</body></html>`

I also tried the same url from inside https://console.pathfinder.gov.bc.ca:8443/console/project/dbc-mapsdk-tools/browse/pods/smk-11-2s6ws?tab=terminal and got same result.

I also tried:
https://mpcm-catalogue.api.gov.bc.ca/pub/mpcm/services/catalog/
https://mpcm-catalogue.api.gov.bc.ca/pub/mpcm/services/
https://mpcm-catalogue.api.gov.bc.ca/pub/mpcm/
https://mpcm-catalogue.api.gov.bc.ca/pub/

same result

**Comment by NicoledeGreef** [comment]
i was messing around with the URLs based on what i saw at the root:
https://mpcm-catalogue.api.gov.bc.ca/

Will one of these URLs do the trick?
https://mpcm-catalogue.api.gov.bc.ca/catalog/PROD
https://mpcm-catalogue.api.gov.bc.ca/catalogV2/PROD
